tezos/sys/src/lib.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+#![forbid(unsafe_code)]
+
+//! This crate exists to handle linking against libtezos.so, this file is empty on purpose.
+
+#[link(name = "tezos")]
+extern "C" {
+    pub fn initialize_ml_context_functions(
+        ml_context_send_action: unsafe extern "C" fn(isize) -> isize,
+    );
+}
+
+/// This function does nothing. It exists to force cargo to link libtezos to crates
+/// that depend on tezos-sys.
+/// To do so, define a `pub` function in the crate that requires libtezos linking that
+/// calls this function.
+/// It doesn't seem to be required under normal compilation conditions, but when
+/// running the tests with the address sanitizer enabled, linking fails without
+/// this extra step.
+pub fn force_libtezos_linking() {}
```

Comment:
I have to expand this comment a bit to explain why it works. The problem is that cargo seems to not link in stuff that doesn't get called (but I have only experienced these issues when running the tests with ASAN enabled, and not always). What I do with this, is to add a call from the crates that require libtezos to be linked (the call does nothing), that way cargo see that the crate is used which forces it to be linked.